src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,75 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		filterableProperties?: NodeFilterProperty[];
+	}
+
+	export interface NodeFilterProperty {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeFilterPropertyDataType;
+		/**
+		 * The description of the filter property
+		 */
+		description: string;
+	}
+
+	/**
+	 * NodeFilterChoiceProperty is used to define the choices for the filter property if the type is choice
+	 */
+	export interface NodeFilterChoiceProperty extends NodeFilterProperty {
+		/**
+		 * The list of choices for the filter property if the type is choice
+		 */
+		choices: string[];
+	}
+
+	export interface NodeFilter {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeFilterPropertyDataType;
+		/**
+		 * The operator of the filter property
+		 */
+		operator: NodeFilterOperator;
+		/**
+		 * The applied values of the filter property
+		 */
+		values: string[];
+	}
+
+	export enum NodeFilterPropertyDataType {
+		String = 0,
+		Number = 1,
+		Boolean = 2,
+		Date = 3,
+		Choice = 4
+	}
+
+	export enum NodeFilterOperator {
+		Equals = 0,
+		NotEquals = 1,
+		LessThan = 2,
+		LessThanOrEquals = 3,
+		GreaterThan = 4,
+		GreaterThanOrEquals = 5,
+		Between = 6,
```

Comment:
@Charles-Gagnon @aasimkhan30 , Between and NotBetween are for numbers and dates, right? I don't think it is needed since user can easily do it by using the less/greater than operators.

Reply:
I like it - makes the UI simpler too since you don't have to support multiple fields on one line and all values are just a single string then.

Reply:
The reason for doing that is the filter dialog will show all the filters at once in a table. I was not planning on letting users add or remove filters from the dialog. The between operator is useful in that UI. 
<img width="643" alt="image" src="https://user-images.githubusercontent.com/6816294/234715517-46061eb8-63f1-4ddf-84b9-f789ebe19825.png">

Reply:
How do you expect to have "Betweens" show up in that example? There's only one value column, but you'll need two for betweens.

Reply:
<img width="634" alt="image" src="https://user-images.githubusercontent.com/6816294/234716410-e4c92d95-a374-492a-a5de-dccfc583c21e.png">

Reply:
And adds another row. This patter is used in SSMS.

Reply:
Hm, well I don't really like that overall from a design perspective but I guess as a initial implementation it's not the worst. I assume that being able to add/remove filter rows is a priority feature for next release then? That's something I imagine a lot of people will be asking for.

Reply:
Both the designs, imo have tradeoffs. This design lets users see all the filters at once and then lets them set or clear filters based on their needs. The other design where users have to add filters one by one works well in the case where we have too many filters. Right now, we do not have too many filters for an object, which will just result in multiple clicks in case of second UI. Adding @erinstellato-ms and @cheenamalhotra for their suggestions on this.

Reply:
The second UI looks like the profiler filter
<img width="640" alt="image" src="https://user-images.githubusercontent.com/6816294/234718382-9034217f-6346-4f3f-abc6-e3fcf65aa101.png">

Also in the second UI the description of the filter will have to be hidden in an info button. So we will be displaying lesser info to the users.

Reply:
We can have this discussion offline - it's not really necessary to nail down for this initial work. But it's definitely something worth thinking about. There's lots of potential improvements here.

Reply:
@aasimkhan30 the filter dialog in profiler was added by me, you can consider reusing it.